### ISSUE TITLE
Error: bad signature

### ISSUE THREAD

**Issue by anjaw3f** [polkadot-js/apps#4487]
A few users are reporting the error "**1010: Invalid Transaction: Transaction has a bad signature**" when trying to send KSM/DOT from their Ledger. 

What does this error refer to? 
Can more context be added to the description?

**Comment by jacogr** [comment]
That gets returned from the Node itself - it means that the signature that is generated by the Ledger for the transaction payload is not recognizable by the RPC node. Not sure what can cause it.

**Comment by anjaw3f** [comment]
Will check with Zondax

**Comment by jacogr** [comment]
Is it widespread atm? Or a couple of users? This is quite weird - and a nightmare to track down in most cases.

**Comment by Tbaut** [comment]
FWIW I've seen this yesterday, with Parity Signer when I took my time (a minute or 2) between getting the payload, and actually signing it.
I did it again and signed right away, and it worked.

I was expecting an error, but I'm used to see something like "transaction expired".. so not sure the longer time I took for signing was the culprit.

**Comment by jacogr** [comment]
That could be it as well - the default mortality is 5 minutes, so basically you don't have unlimited time before the transaction becomes invalid.

**Comment by MirekHruska** [comment]
Hi, few members of our community had this problem too. This is caused when you using Ledger with Kusama and Polkadot apps. In new polkadot.js.org version, account are moved accross. When you create account on polkadot and switch tu kusama network, there is new account created with kusama format address. But when you deposit that account, you can't access it to move you funds (Error: bad signature is shown). Here is video https://youtu.be/RIT5uyfDp-c  and here is full video (but in czech language) https://studio.youtube.com/video/KxWrc3S4Q4E . You need to tag that accounts to be sure you will not deposit bad account.

**Comment by jacogr** [comment]
Ok, so make sure you attach the account (tie to this network) in the settings. Initially when the always-show-Ledger enhancement went it, it didn't tie it to a specific chain. 

The UI does do that at this now since the start of last week - but for accounts that were added before, you need to click the below toggle under the 3 ellipsis, so it only shows on the specific chain.

![image](https://user-images.githubusercontent.com/1424473/105706434-d2c2da00-5f11-11eb-92a5-6104689dbb4c.png)

**Comment by anjaw3f** [comment]
Jaco: So in order to prevent this from happening, users should make sure that "only this network" is switch ON, yes? 

Zondax team: For all the others who this has now happened to - are their tokens lost? Or is there any way to retrieve them? My understanding is that the automatically created KSM address is on their Ledger, so shouldn't they be able to access it? Or am I mistaken?

**Comment by MirekHruska** [comment]
@jacogr thanks, but I think this is serious trap for novices. Is there solution for allready deposited accounts? You can't use polkadot app on kusama network and if you try to sign your transaction with kusama app, bad signature error will appear. Are those kusama (or dots) lost forever? There are people with hundreds dot/ksm locked on that accounts.

**Comment by jacogr** [comment]
Just tie the account to the network, so it only displays. Or remove/re-add and it will by default tie.

The account visibility is within user-control here.

**Comment by MirekHruska** [comment]
@jacogr as I said, there are a lot of people that allready send their dots/ksm to that accounts. On youtube, there is a lot of instruction videos that didn't mentioned this and a lot of novices will fall into that trap in near future. So, if someone ask me if his dot/ksm is lost forever, answer is yes? Or there is solution? Thanks

**Comment by jacogr** [comment]
Since accounts are always derivable by the mnemonic and the account/address indices, nothing is ever lost since you can always get the private keys based on that info. 

If you have access to the seeds & offsets and it is on a device for a specific network, you can get to the funds. (The addresses are just the publicKeys in a human-readable format, if you have the mnemonic you can get to the same private key which will yield the same public key)

**Comment by MirekHruska** [comment]
@jacogr thanks. But there must be some difference, no? When I add my account on polkadto network (Ledger 0/0) and switch to kusama network and add same (Ledger 0/0), it generate two different accounts. So, solution is put memoric seed from ledger (24 words) and restore account on polkadot network from it, switch to kusama network and send KSM out?

**Comment by jacogr** [comment]
No, the accounts are the same - what you are seeing is the ss58 address, when that is decoded it would yield exactly the same public key.

**Comment by MirekHruska** [comment]
@jacogr are you sure that accounts are same. Account Ledger 0/0 moved from Polkadot => Kusama network has different public addres than same account (0/0) directly added via add ledger button on kusama network with kusama application on Ledger device. There must be some difference, becaus if doesn't matter where you created your account, this thread shouldn't exists :-)

**Comment by jacogr** [comment]
See the example above :) ss58 is only a display format. It encodes the network ss58 + public key + checksum into base58. On decoding you stip all of that and only use the public key. when sending to the node, the ss58 is not transferred, the actual public key (which is the AccountId) is.

**Comment by MirekHruska** [comment]
@jacogr ok, thanks, I will try to study it more detailed. Good news is that there is solution, superb. Thanks again.

**Comment by jleni** [comment]
> Hi, few members of our community had this problem too. This is caused when you using Ledger with Kusama and Polkadot apps. In new polkadot.js.org version, account are moved accross. When you create account on polkadot and switch tu kusama network, there is new account created with kusama format address. But when you deposit that account, you can't access it to move you funds (Error: bad signature is shown). Here is video https://youtu.be/RIT5uyfDp-c and here is full video (but in czech language) https://studio.youtube.com/video/KxWrc3S4Q4E . You need to tag that accounts to be sure you will not deposit bad account.

Unfortunately this is not related to the Ledger app themselves.. but the way the integration works in Polkadot.js. 
I wonder why there is even this option "only this network".

@anjaw3f Would it make sense that polkadot.js adds a popup with an explanation to avoid issues being mislabeled as a problem in the apps?

**Comment by anjaw3f** [comment]
This is good news! Do you have some practical instructions/steps for regular (non-techy) users to access their funds? 
Apparently they can't access them right now, even though their Ledger is connected and generated that address. 
What do they have to do in order to send the funds out?

**Comment by anjaw3f** [comment]
> @anjaw3f Would it make sense that polkadot.js adds a popup with an explanation to avoid issues being mislabeled as a problem in the apps?

I'm always for anything that makes things easier and more clear for users. We do have a lot of beginners, so the clearer we can spell things out for them the better.

**Comment by gorgos** [comment]
> This is good news! Do you have some practical instructions/steps for regular (non-techy) users to access their funds?
> Apparently they can't access them right now, even though their Ledger is connected and generated that address.
> What do they have to do in order to send the funds out?

I second this as I'm in the same boat. Any instructions how to access the funds are greatly appreciated.

**Comment by ohhichase** [comment]
> > This is good news! Do you have some practical instructions/steps for regular (non-techy) users to access their funds?
> > Apparently they can't access them right now, even though their Ledger is connected and generated that address.
> > What do they have to do in order to send the funds out?
> 
> I second this as I'm in the same boat. Any instructions how to access the funds are greatly appreciated.

3rded

**Comment by elfif** [comment]
Hello i'm upping up that issue that i found it being critical.
I have exactly the same issue as previously described, some KSM coins are stuck on a wallet which address is just a translation of my polkadot address, both wallets use Ledger apps on polkadotjs.
I agree with @jleni that technically issue is more on the ledger integration on polkadot-js than on the ledger itself.
Anyway we need an answer to that issue

**Comment by ohhichase** [comment]
> Hello i'm upping up that issue that i found it being critical.
> I have exactly the same issue as previously described, some KSM coins are stuck on a wallet which address is just a translation of my polkadot address, both wallets use Ledger apps on polkadotjs.
> I agree with @jleni that technically issue is more on the ledger integration on polkadot-js than on the ledger itself.
> Anyway we need an answer to that issue

I've tried everything.  My ksm is gone.  30+ hours with no hope of a result.

**Comment by MirekHruska** [comment]
I'll try to look at it on weekend but right now, there is no step by step instructions. @jacogr , is it possible to create step by step instructions how to set those KSMs and DOTs free? Thanks.

**Comment by jacogr** [comment]
Just to summarize the comments above -

- If you have an Ledger account visible on multiple networks (the UI will show an indicator for those in the accounts list), just apply the "only this network" toggle. The warning will go away. 

- Newly added accounts will always tie to a specific network


If funds have been send to an address on a network where it now yields a signature error - 

- Derivation is open standard, this means the keys can always be derived from the mnemonic + cointype + account + address (the first 2 from the app, the second two as user-selected)

- For this, we are looking at a couple of options (couple being 2) to (using the standard derivation above) ensure you can actually transfer it out (and also being able to recover keys at anytime in the future)

- If there is an update on the previous point, will confirm - as of now just be aware that is being looked at for the best approach(es) by multiple people


If unsure about anything -

- This is not a bug in the Ledger app itself, rather due to the app being used on the wrong network (which the first point above covers)

- Ensure that you trust & verify, i.e. use the option in the account menu to display the address on the device - it should match if the app can sign it correctly

**Comment by ohhichase** [comment]
Note the ledger show a different address than ksm account.  I've recreated. Address shown on ledger but no possible way to get same address thats on "lost" account.

**Comment by elfif** [comment]
Hello @MirekHruska and @jacogr  and thanks for your answers.
If you need any additional informations, or want me to perform some tests of any kind i'm all yours.

**Comment by daledenton1** [comment]
I'm also in this boat. I've just Bonded 50 KSM but when I nominate my validators and click sign and submit I get the 1010  Invalid transaction. Transaction has a bad signature. 
I'm also a novice at this and do not understand all this technical stuff. Really needs to be an easier fix.

**Comment by daledenton1** [comment]
When I check 'only this network' tab, then when i switch between Dot and Ksm the accounts are then missing until I uncheck the tab.

**Comment by ohhichase** [comment]
> When I check 'only this network' tab, then when i switch between Dot and Ksm the accounts are then missing until I uncheck the tab.

ill kinda break it down for you  to where we understand.  ive been at it for over 40 hours now.. the account you put the ksm on is actually your polkadot account, and you have no access to the ksm side of it it just sits there and cant sign anything, if you look at ledger its a wrong address, you can recreate the address on the seed, but not the one thats holding the ksm ... just the correct address that displays on the ledger.  from what i can tell theres no way to get it off the ledger itself without knowing as much as these guys.  This is my first time even using a ledger and attempting to use polkadot-js im mentally exhausted as im sure you are...

**Comment by daledenton1** [comment]
Yeah I've been in a pretty bad mood all day over this issue. Very frustrating indeed!

**Comment by daledenton1** [comment]
Just chatting on Polkadot Watercooler. If you click forget account and then add your key again it should fix it. Haven't tried as yet??

**Comment by elfif** [comment]
I tried, it did not fix anything.
Hopefully, because it would mean that you can also potentially a working account created with the Ledger.

From a noob perspective to make it work would require the "import from ledger" feature to accept keys from both DOT & KSM ledger apps when I try to connect to KSM.

**Comment by ohhichase** [comment]
Ive tried everything.  i derived  hundreds of accounts, even got the one that it says on the ledger when checking the ksm dot accounts....  nothing cant sign with it either, tried proxy, multisig, the offline sig, i tried every single wallet, every possible hard/soft address theres no way to do it that i can find.  i just wanted to get that ksm off of it bad.   i definately need a new seed phrase after the dark back allies i took that one down.

**Comment by elfif** [comment]
Ok i agree @ohhichase issue is nasty, there is real money on the line and i did not think neither of the person impacted really did something wrong or reckless at any moment.
On my side i manage to display the KSM wallet account that has an issue from a second computer. 
As said before that's quite easy in fact. I understand it may be difficult for a dev of the project to catch uyp so let me replay that.
You just need to import the DOT wallet account from your ledger (the one from which your problematic KSM account is derived).
Then you just need to switch off the "only on this network" switch in the account menu (see screenshot)
Then switch to KSM network. Your KSM account is there with the expected KSM balance.
But issue remains, you can't sign any operation because it would require signing using Ledger DOT app and the polkadot-js app expects only TX to be signed with the KSM one....

**Comment by daledenton1** [comment]
I tried that last night. Switched to only on this network on DOT and my KSM wallet wasn't there. Switched it back and my KSM balance was back. Still unable to sign transactions. Very frustrating!

**Comment by MirekHruska** [comment]
I tried to solve this today but I totally burned out. I think that solution will be extract account out of ledger straight to polkadot.js.org account, but I don't know how to recover account without using ledger. There is a lot of topics on different memoric seed formats and compatibility, a lot of linux command line tools etc ... nothing what normall user can use. I know that it's possible to get that account out of ledger? @jleni is it possible to write step by step instructions how securely import polkadot / kusama account from ledger straigth to polkadot.js.org? Some simple offline tool (even linux / windows command line) that generate polkadot JSON file based on Ledger BIP39 memoric would be nice :-) Thanks

**Comment by ohhichase** [comment]
i just had a small breakthrough but too tired to finish.... got a new phone today and put trust app on it.  imported a polkadot account with qv and checked later to show no balance.   I believe only change i made was on polkadotjs and changed the address prefix from kusama to polkadot in the polkadot portal.....  now im not sure exactly how id attempt because trust doesnt have a kusama wallet, but i think its an answer to pulling the kusama off if i can get all stars to align.   ill get on it tomorrow... 

note i have zero experience in this.. im not a tech guy or even a computer guy at all... im just straight trial and error on it in the best ways i can assume work.  its a learning curve for sure.

**Comment by ohhichase** [comment]
> I tried to solve this today but I totally burned out. I think that solution will be extract account out of ledger straight to polkadot.js.org account, but I don't know how to recover account without using ledger. There is a lot of topics on different memoric seed formats and compatibility, a lot of linux command line tools etc ... nothing what normall user can use. I know that it's possible to get that account out of ledger? @jleni is it possible to write step by step instructions how securely import polkadot / kusama account from ledger straigth to polkadot.js.org? Some simple offline tool (even linux / windows command line) that generate polkadot JSON file based on Ledger BIP39 memoric would be nice :-) Thanks

ive tried about every way to derive the file onto polkadotjs ive tried every address ie m/44'/434'/0'/0'/0' in every combination on every single app that gives the option to choose the derived path.   I think if i can get the kms account that has the same address as the polkadot kms account again and get that private key i might be able to find a multiwallet with both that could work.. ill update.

**Comment by MirekHruska** [comment]
https://github.com/paritytech/substrate/issues/7824 there si topic about Ledger support in substrate subkey (still opened). Seems that there is no easy way right now. I think that after more study, I'ill figure it out, but there is no way how to make instructions for ordinary nontechnical people. Everything is too 'geeky'.

**Comment by Tbaut** [comment]
You're not alone in this. Zondax is working with Polkadot.js to make a recovery possible. Please be patient and prevent from spamming this issue. The problem is understood and a workable solution is on its way.

**Comment by jleni** [comment]
Yes, correct. 
We are working in coordination other stakeholders.

In particular, we are building a special recovery tool (based on a Ledger app) that will allow moving funds across chains to mitigate these problems that people are having when using Polkadot.js

>  @jleni is it possible to write step by step instructions how securely import polkadot / kusama account from ledger straigth to polkadot.js.org?

We also aim to provide additional tooling/libraries to generate keys but we VERY STRONGLY advise against entering your mnemonic in web or deskop apps.

Early next week we will have a more comprehensive article and a FAQ section about this topic.

We are very committed to this community and do not worry if you sent fund to your own (but incorrect addresses) funds are safe and the recovery tool will be available ASAP.

**Comment by jleni** [comment]
Just for clarity:

- we DO NOT recommend entering your Ledger mnemonic in any tool beyond the device itself. 
- we will prepare a Ledger-based recovery tool in the next few days that will allow people to recover funds sent by mistake to public keys in other substrate-based  chain.
- we will keep you updated here and in social media.

**Comment by gorgos** [comment]
Just posted this in Discord as more and more people are running into this issue and it will be interesting for anyone following here as well:

In my opinion this was a serious UX bug by polkadot.js.org, it shouldn't have happened at all. The fix of adding exclamation marks is in my opinion not good enough, people may still send money to this address.

For anyone curious about the technical details, when you create an account with Ledger on polkadot.js.org for Polkadot, a public key is generated by the Ledger device using the Polkadot app. The ledger app has built-in configuration for how to derive private keys from the mnemonic. Following the BIP-44 standard it's correctly using 354 in the derivation path, see https://github.com/satoshilabs/slips/blob/master/slip-0044.md.

When you now switch to Kusama, the previously created Polkadot account is for some reason also shown in the UI. If you now send KSM to this address and later try to transfer, it will throw 'bad signature'. This is because the Kusama ledger app is using 434 in the derivation path and so the generated private keys are different. When you add new accounts in Kusama via Ledger, you will see that a different account is created for the same account index.

There's no easy solution for this, but the funds are not lost. You can either insecurely important the mnemonic and use a newly created recovery tool to obtain the private key. Not recommended, so I won't post details here, but I solved the issue for me this way.

Or there needs to be direct support by the Ledger apps which is already in the making. So just stay tuned and follow the Github issue.

**Comment by feelwelltm** [comment]
5 days and nothing is happening,when i can move my kusama???? when we will prepare a Ledger-based recovery tool????? when I have to be able to move my kusama in any account I want !!!!!!!!!!!!! hello !!!!!!!!!!!!! solve the problem !!!!!!!!!!!!

**Comment by daledenton1** [comment]
They are hoping to have a fix possibly within the week, so maybe another few days.

**Comment by feelwelltm** [comment]
nothing has happened yet...........

**Comment by ohhichase** [comment]
When I apply to kappa sigma mu i hope my patience will be taken into
consideration.  Lol.... but I'm serious i think.

**Comment by daledenton1** [comment]
Yes an update by now would be nice...

**Comment by elfif** [comment]
Agreed.
I'm usually very supportive regarding open source. But there is some real money on the line.
I send my KSM to that wallet in order to stake it and i haven't been able to. It's been more than 2 weeks now.
In the end your inaction is keeping some people money frozen. 
That's just bad.

**Comment by feelwelltm** [comment]
> That's just b

yes they have to move !!!!!!

**Comment by feelwelltm** [comment]
from ledger no answer from 2 emails a week ago!!!! trash support from ledger

**Comment by jacogr** [comment]
As the ZondaX stated above, they are working on a recovery app which is the preferred option since the mnemonic never leaves the device. Additionally as they said (this team does the apps for Ledger for Polkadot & Kusama) anything else but the recovery app is not something they endorse nor would recommend. 

Similar how crypto wallets in other ecosystems allow you to import Ledger-derived accounts directly, there are 2 other options already available which allows you to derive a compatible account and use that seed. Neither of these other options have been announced (although people following GH are aware of them), the lack of any announcement is specifically to allow for the first-prize option to be completed and to be available on the Ledger store. 

The first non-recommended alternative is a command-line tool that can derive the ed25519 key from the mnemonic (already used by some in this thread on non-primary computers), the other is using the create account on the apps UI with the Ledger-compatible crypto option. 

The primary objective for these are that you are not just dependent on one vendor implementation i.e. there are valid reasons to want recovery on another device (so it is not completely back-boxed and certainly vendor-neutral), however the ability to do this is also why there are so many "enter your mnemonic" phishing sites out in the wild.

TL:DR The best is waiting for the Ledger app from the ZondaX team, they are working hard on it, there are alternatives if you are willing to follow a strongly-not-endorsed approach.

**Comment by elfif** [comment]
Hello @jacogr  and thanks for your answer that is putting some perspective.
First i bought a ledger to avoid entering my mnemonics, so i do not consider the alternatives you are talking about and i think they should not be disclosed.
Second i understand the issue is "in between"  Zondax team and polkadot-js team, this thread is in polkadot-js so maybe we should more complain on the KSM ledger app Github repo regarding delay.
On the other side the thing that scares me even more than my own case is the fact the issue is still out there, i mean i just tested polkadot-js and apart from a vague warning the option to make an account visible on all network is still present so basically even today someone could still end up in my situation, which is fully unacceptable.
This kind of situation makes me wonder about who is the pilot in that plane. Even before thinking about how to help us recover our KSM the first priority should have been to stop the bleeding and not just by providing a warning.......

**Comment by jleni** [comment]
We've been definitely working hard on this special recovery mode to allow people retrieve KSM that were sent to incorrect pubkeys. 

The code is ready and we are talking to Ledger to streamline the review process and publication.

[Kusama App - DOT recovery mode](https://github.com/Zondax/ledger-kusama/pull/64)

PLEASE DO NOT USE THIS CODE. THE APP WILL BE PUBLISHED VIA THE OFFICIAL CHANNELS SOON.

In the meantime, we are working on some detailed instructions once the app is published to mitigate any other user mistakes. 

I agree that it would be important to avoid this from happening at the UI/UX level. In particular, we don't think this "recovery mode" should be kept as part of future releases and we will deprecate and remove it in the near future.

**Comment by jacogr** [comment]
The warning was not added for Ledger. Ledger accounts are not handled in any other way than other accounts. It was specifically added so people are aware of which accounts are shared between networks, most users have the bulk tied to networks, but some specific ones not. (Same reason why dev accounts are marked in a similar fashion)

The apps UI does not do handholding, doing so is not a focus. It is certainly not the simplest solution out there and it is not meant to be, the primary focus is on making it the most comprehensive advanced tool. I know this is not the expectation, but it is the reality of where focus is going.

There is lot that can and should be done. Around accounts there is a grant being worked on that has been awarded to a team that will finally bring some streamlining into at least the accounts sections (catering for non-advanced users that need guidance).

**Comment by elfif** [comment]
@jleni thanks for the update. 
I also posted on The KSM Ledger app issue, Zondax should update status there aswell....
My question is : "Do you have any idea how much people are impacted ?" Some users may have just deposited some KSM on the wrong address and never tried to access it since.  We are only a few people complaining here but the issue was so easy to encounter you may have way more cases in the wild.....

**Comment by MagnatUK** [comment]
I hope this will be fixed. It is auctions time soon and funds have to be freed ;]
Hope Ledger will be quick. Pls update here asap. thx

**Comment by feelwelltm** [comment]
UES YOU ARE RIGHT !!!!!!!!!!!!😎

**Comment by feelwelltm** [comment]
WE NEED THE KUSAMA FREE TO MOVE AROUND !!!!!!!!!!!!!🤐

**Comment by feelwelltm** [comment]
NO ANSWER FROM LEDGER YET???? I CANT EVEN STALKING WITH POLKADOT FROM LEDGER DO SOMETHING,AND MY KUSAMA ARE BLOCKED IN THE DOT ADRESS !!!!!!!!!

**Comment by daledenton1** [comment]
Nothing as yet! I've got over 100 KSM and 660 DOT that I can't do a fucking thing with!! They were hoping for it to be fixed within the week which was last week. This is real money, not a fucking game! This should not have happened! I'll be taking both my KSM and DOT out of Polkadot.js as soon as this is fixed. I've since found Kraken does staking for DOT and KSM that's a lot more straightforward than the polkadot.js crap!!

**Comment by Eightill** [comment]
I'm an average retail investor with the same problem. Getting the 1010 error when trying to withdraw KSM tokens and a "bad signature" error when trying to withdraw Edgeware tokens. 

To the folks working on this issue: do you think the Edgeware errors are related to the KSM/1010 error? 

Also, I'm not going to be like feelwelltm above and yell in all caps and use a hundred exclamation points, but this IS seriously inconvenient. Will you kindly post an update on this? Thank you!

**Comment by feelwelltm** [comment]
first error Spec version not supported. second error 1010,all my dot and Kusama are blocked I can't do a fucking thing with! polkadot.js is really a trap,really this is a game for your company with our real money???? daledenton1 you are 100% right...who knows how many people have the same problem and even don't know.   Eightill this are my savings and is a big amount not a fucking game like daledenton1 is saying. WE WHANT THE MONEY FREE TO MOVE!

**Comment by MagnatUK** [comment]
Any update from Ledger ?

**Comment by elfif** [comment]
In my opinion guys we are complaining on the wrong place.
I have some KSM stuck also, just to be clear.
A fix is on this way on the ledger KSM app repository.
Issue is declared there too : https://github.com/Zondax/ledger-kusama/issues/56
Development to have a recovery mode is done too : https://github.com/Zondax/ledger-kusama/pull/64

Now i'm not sure what is the current status of that process. I think they have to build a new version of the app containing that fix, test it intensively, submit it to ledger, wait for validation and prepare a documentation. 

They are careful and i agree with that, even if it takes time, no one wants the cure to be worse than the poison.

**Comment by MagnatUK** [comment]
I was given this website by polkadot support. They said it should be fixed this week. We have Friday ... as long as it will be fixed before kusama auctions I can survive but losing on possible staking rewards ...

**Comment by feelwelltm** [comment]
in the ledger today was an update with kusama,I have made the update in my ledger but the same error 1010....offffff

**Comment by gorgos** [comment]
> in the ledger today was an update with kusama,I have made the update in my ledger but the same error 1010....offffff

See my comment [here](https://github.com/Zondax/ledger-kusama/pull/64#issuecomment-778490880), maybe try the second app version.

**Comment by elfif** [comment]
Hello there. 
@gorgos =>I successfully installed the kusama_sec app.  
@feelwelltm @daledenton1 => you need to enable "Developer mode" in settings => experimental feature of ledger live.
Then you will see the app called kusama_sec.

**Nevertheless, issue remains on my side.**
I went to polkadot-js and i follow those steps:

- Switched to **on** the "only for this network" radiobox on my polkadot account that is managed by my Ledger
- Went to Kusama network and removed all accounts, just in case.
- Launch the kusama_sec on my ledger
- Import account from my Ledger

And then yes an account appeared, but it's empty. **My KSMs are not there**.
This app generated yet another account with another public address.
But it's still not the good one.

I have no idea how this thing got tested or even if there is only one potential issue regarding interactions between polkadot-js and Ledger.........
I do believe this kusama_sec app might fix the issue for some but it fixes nothing in my case.
Or maybe i'm not using it correctly, which is totally possible as no one issued instructions regarding how to use it.

**Comment by gorgos** [comment]
@elfif Going from the referenced PR: https://github.com/Zondax/ledger-kusama/pull/64/files#diff-23182787290de2724918bd148338edf92d101964ded597c540736d050a831f70R30-R34, you should be seeing a 'You are about to enable the DOT recovery mode.' message. You'll have to activate this somehow, I would look in the Kusama Ledger app settings on your device.

**Comment by elfif** [comment]
@gorgos, i checked already, nothing there apart enabling expertmode, which i did and it changes nothing

**Comment by jleni** [comment]
As explained here https://github.com/Zondax/ledger-kusama/issues/66#issuecomment-778172779 the last release was NOT about recovering KSM sent to Polkadot pubkeys.

> As previously described, this is a very specific case (incorrectly sending coin to pubkeys in other chains) requires a recovery feature that is not standard and involves multiple stakeholders and a more complex process to get released.
It is very unfortunate for affected people. We have been asked to contribute with a recovery solution for this community issue and several people in our team (in addition to other stakeholders) are working on a tool to help everyone affected.
>
> We will soon provide access to this tool, videos, tutorials, etc. but we must follow the right process.

**Comment by dartollen** [comment]
Could we please have a status update? Thanks very much.

**Comment by daledenton1** [comment]
Anything??????

**Comment by elfif** [comment]
The way this issue is handled is a massive joke. 

@jacogr you said 14 days ago that user handholding and so on was not in the actual priority for this project , like basically you do not care, this project in its actual state allows users to get their money stuck without receiving any warnings and it's normal.....
@jacogr  & @actions-user you seems to be the 2 devs really in charge here based on commit activity.
You know during those last 14 days some more people got some money stuck because YOU DID NOTHING to warn them ?
But in the same time you kept comitting some other stuffs to polkadot-js, because yes, this is a wallet after all, so allowing people to use it to keep their money safe is not a priority....

This thing is a joke

**Comment by jacogr** [comment]
The "actions-user" is a CI bot.

You can already re-create your account using the apps UI, so there is nothing that stops extraction and indeed it an avenue taken by people already that didn't want to wait on a Ledger recovery app.

The warnings for unassigned accounts are in the UI.

**Comment by Eightill** [comment]
Still getting the 1010 error also. WTF Kusama?!?! This is seriously BS.

**Comment by daledenton1** [comment]
Is there a tutorial or a walk through that is available for this process? Or is there someone willing to do one? I want out!!

**Comment by carlmcconnel** [comment]
So this supports Ledger accounts natively for a couple of weeks now. So not sure why people are swearing and waiting on a "Ledger recovery app" when you can just add the account anyway if desperate don't think it is good to need to have a Ledger to get access to funds for instance of Ethereum I can import my Ledger accounts without the Ledger.

Did this for mine to create additional peace of mind after I asked https://github.com/polkadot-js/apps/issues/4668 - there was also somebody commenting there on this issue where he unlocked.

- go to accounts
- remove the account
- use the + account
- selected advanced to get to the Ledger type option
- selected correct app and the account and address numbers
- check address against expected
- create
- use as normal

if you sent on the DOT network but only have a KSM app you obviously connect to polkadot and when creating select the kusama app or the pother way around. Hence the check against the address. 

Once transferred, you can just remove the account and add your Ledger back - this is what I did after making sure I can use the account in the app and in the extension. Anyway for me after i know it is working feel safer now so if something does happen I always can always create the raw account and have access - as long as I have my seed.

**Comment by dartollen** [comment]
The issue is not that people can't add Ledger accounts. The issue is that people added a DOT Ledger account, then switched networks to KSM. That same DOT account is then auto-translated by polkadot.js.org into KSM format. However, if you send KSM to that translated KSM address, you can't recover it because you can't sign transactions using the KSM app on the Ledger. If you add a new KSM account from Ledger using same address type/index 0/0, it results in a completely different wallet address from the translated DOT address. People are very frustrated that they have significant amounts of money locked in accounts they cannot access. Solutions have been promised but not provided and no substantive updates have been given for weeks now. If there is some method of extracting via polkadot.js.org, much clearer instructions are needed.

**Comment by carlmcconnel** [comment]
That is exactly what the account import allows so you can add the translated address with the correct keys. In you example is the same as my steps that i set above

- connect to kusama, 
- go to accounts,
- + account, 
- insert mnemonic, 
- select advanced options,
- select type to be ledger, 
- select polkadot type app (your example had ksm sent to dot app), 
- create, 
- use

**Comment by dartollen** [comment]
Thank you for the explanation - I appreciate that. The problem is that this solution involves the mnemonic leaving the Ledger. This defeats the whole purpose of using the Ledger in the first place and jeopardises the security of all accounts in all cryptocurrencies stored on that Ledger. It's good to know that there is a possible workaround, but it is not a solution which is compatible with the concept of using a hardware wallet.

**Comment by carlmcconnel** [comment]
Understood. I read above and my soliution is just 1 of the 3 suggested, there seems to be a ledger app one on the way.

**Comment by jleni** [comment]
The upgrade to both apps got accepted.. they should be published by Ledger very soon:

https://github.com/LedgerHQ/app-kusama/pull/7
https://github.com/LedgerHQ/app-polkadot/pull/4

Blog and videos are also ready to go! Stay tuned

You can find a bit more info here: https://medium.com/zondax/introducing-ksm-recovery-mode-465a1a30e6df

**Comment by dartollen** [comment]
Thank you for the update, Juan - much appreciated.

**Comment by ohhichase** [comment]
I'm throwing my ledger away.  I just would like access my ksm.   Please let
me know if there's any way to access.  There's been a lot of people
"knowing what they're talking about" yet don't in this thread.

**Comment by daledenton1** [comment]
Very soon never doesn’t seems to be that soon!

On 23 Feb 2021, at 8:07 am, ohhichase <notifications@github.com> wrote:

﻿

I'm throwing my ledger away. I just would like access my ksm. Please let
me know if there's any way to access. There's been a lot of people
"knowing what they're talking about" yet don't in this thread.

**Comment by daledenton1** [comment]
The Ledger is a piece of junk! Doesn't even support most of the tokens I own. It's caused me nothing but headaches!

**Comment by ohhichase** [comment]
Doesn’t work, say its x / y when you create it the way we all did it gives you x / x and says x / y as the addresses on the ledger.  You cannot recreate it as x / x and that’s where the ksm is.  Im able to reproduce the y account, which is what the ledger reads under my address, cannot recreate the x account as ksm.  Does that make sense?  Sent from Mail for Windows 10 From: Jaco GreeffSent: Monday, February 22, 2021 2:36 AMTo: polkadot-js/appsCc: ohhichase; MentionSubject: Re: [polkadot-js/apps] Error: bad signature (#4487) The "actions-user" is a CI bot.You can already re-create your account using the apps UI, so there is nothing that stops extraction and indeed it an avenue taken by people already that didn't want to wait on a Ledger recovery app.The warnings for unassigned accounts are in the UI.—You are receiving this because you were mentioned.Reply to this email directly, view it on GitHub, or unsubscribe.

**Comment by ohhichase** [comment]
its been 28 days, i see theres an update, it says the solution was to be released in a couple hours, but i dont see any way to do it.   i need to remove my ksm, and sell it to make up for everything else in the market, please someone stop what they are doing thats more important than thousands and thousands of our money likely hundreds of thousands and explain this... my ksm isnt staking, this is 28 days of this.. one of you have to understand the urgency, and for whatever reason we get blamed for doing it when its a pretty shitty design maybe the use only on this network should explain what that does.   anyone want to help?  ill give you the fucking 24 word phrase at this point, i just want to be out of this wallet.

**Comment by daledenton1** [comment]
They think everyone is a fucking computer engineer and can just figure this shit out! One wrong click and goodbye to my $$$

**Comment by gorgos** [comment]
Guys I understand your frustration, but you're really complaining at the wrong place about the wrong things.

The initial issue: This has never been an issue with Ledger itself and not even with the separate company developing the Ledger Polkadot/Kusama app. They worked as expected and correctly. The issue was a UX problem inside polkadot.js. And as it seems it's only one guy working on this, so you can't really blame him either. I think it would be good for Polkadot to allocate more funding towards polkadot.js. Add a few more devs and a UX designer.

The long wait for the fix: This has never been an issue with polkadot.js, nor the Ledger app devs. They added a fix pretty quickly, but I guess it takes time for Ledger to review apps. And you can't really blame Ledger either for doing proper app reviews. If they didn't, it would jeopardize the security of all users.

So again, no point complaining here. Send a message to the Polkadot support asking that polkadot.js should become more user-friendly. And don't forget no funds here were ever lost, you just have to wait.

**Comment by daledenton1** [comment]
So it's no one's fault then?!

**Comment by gorgos** [comment]
> So it's no one's fault then?! My funds just got stuck but hey I must have caused it!

As I said, it's not your fault, but the core issue was Polkadot has missed that they are a number 4 coin now and completely gone into mainstream despite not even doing that much marketing. And with the publicity come users that require an easy to understand interface. Polkadot has such amazing dev support and documentation and tech, so now it's time to also make this more accessible to end users.

**Comment by ohhichase** [comment]
dude, give us the way to get it with our phrase so it can be on us.   we want a solution this underdeveloped piece of shit app shouldnt have ever been put on ledger.   period.   dot and ksm should be taking this a little more seriously.

**Comment by ohhichase** [comment]
ksm is the only coin up.. i need to relocate that money.  or break open treasury and compensate

**Comment by daledenton1** [comment]
Yeah would have been nice to have access before this massive fucking crash!

**Comment by gorgos** [comment]
> dude, give us the way to get it with our phrase so it can be on us. we want a solution this underdeveloped piece of shit app shouldnt have ever been put on ledger. period. dot and ksm should be taking this a little more seriously.

Instructions were already posted here: https://github.com/polkadot-js/apps/issues/4487#issuecomment-783344215. With the note that this should only be done if you have only KSM on Ledger and afterwards reset it to a new mnemonic.

**Comment by ohhichase** [comment]
so the dot i have 5 days left unstaking makes it where i cant do it?   this is definitely a fuck up someone isnt admitting to.    in no way would something be designed like this.

**Comment by avdende** [comment]
Hello everyone, 

The KSM and the DOT Recovery Modes are operational on ledger devices. 

**DOT Recovery Mode**:

- [Written instructions](https://medium.com/zondax/how-to-recover-dot-sent-to-the-wrong-network-kusama-2f89b5d0b93d)
- [Video tutorial](https://www.youtube.com/watch?v=qJPwguvRz80)


**KSM Recovery Mode**:

- [Written instructions ](https://medium.com/zondax/how-to-recover-ksm-sent-to-the-wrong-network-polkadot-f8865fa094a0)
- [Video Tutorial ](https://www.youtube.com/watch?v=hM0ApWzEzeA&feature=emb_title)

**Comment by dartollen** [comment]
This worked and I have recovered my KSM - thank you very much for sorting this out. Really appreciate it.

**Comment by elfif** [comment]
Same here, Thanks a lot to all the devs that made the fix possible.

**Comment by daledenton1** [comment]
So I've recovered my KSM and DOT funds and transferred them out of Polkadot.js but now my bonded funds in both KSM and DOT have disappeared! This thing is an absolute nightmare! Anyone care to shed some light on where these funds might be. They were both bonded funds but I was unable to stake them due to the signature issue.

**Comment by daledenton1** [comment]
My balances are visible when I enter the addresses into Polkascan. On Polkadot.js my KSM appears in account actions in staking but I get the 1010 Bad signature still when I try to unbond it? My bonded funds for DOT are nowhere. Is there something I've overlooked or is this going to be another issue?

**Comment by daledenton1** [comment]
So I've unbonded both KSM and DOT. You have to uncheck the view only on this network tab. So after the bonded period is up, am I going to be able to send my funds out of there? Took about 10 seconds to stake on Kraken! And I didn't need to have a Phd in computer engineering.

**Comment by MagnatUK** [comment]
Doesnt seem to work on LedgerX

Is it only for Ledger S

Kris

**Comment by daledenton1** [comment]
Yes it works with a Ledger X. I was able to retrieve and withdraw both my KSM and DOT funds on my Nano X. I'm just having trouble with my bonded funds.

**Comment by osorio200** [comment]
I have the same error but don't understand a single thing you people are saying. Can't get my KSM out.

**Comment by osorio200** [comment]
I understand it may be caused if the KSM is in a DOT address?
How can I check if the address my KSM are in is a DOT or KSM address?

**Comment by MagnatUK** [comment]
Managed to do it as well.
THX TEAM !

**Comment by osorio200** [comment]
Guys, I will give $100 to whoever can help me get my funds out. If you're interested, contact me at osorio200@yahoo.com

**Comment by jleni** [comment]
> Guys, I will give $100 to whoever can help me get my funds out. If you're interested, contact me at [osorio200@yahoo.com](mailto:osorio200@yahoo.com)

The instructions+video here should be enough for you to do it... 

The KSM and the DOT Recovery Modes are operational (on ledger devices). This recovery mode allows you to unblock and recover KSM that got accidentally sent to Polkadot addresses, and vice versa (retrieve DOT that got accidentally sent to a Kusama address).
 
**DOT Recovery Mode**:

* [Written instructions](https://medium.com/zondax/how-to-recover-dot-sent-to-the-wrong-network-kusama-2f89b5d0b93d)
* [Video tutorial](https://www.youtube.com/watch?v=qJPwguvRz80)

**KSM Recovery Mode**:
 
 * [Written instructions ](https://medium.com/zondax/how-to-recover-ksm-sent-to-the-wrong-network-polkadot-f8865fa094a0)
* [Video Tutorial ](https://www.youtube.com/watch?v=hM0ApWzEzeA&feature=emb_title)

**Comment by osorio200** [comment]
Hello Juan,
Thanks for your reply. I have still not been able to fix this.
While I tried to fix it, In the Polkadot network I clicked on “Forget this Account” because it was unused and had 0 DOT, and when I went back to the Kusama network, it turns out it was the Kusama address that had the KSM, and it was gone.
I have the address, but I do not have the pnemonic phrase. Do you think it is possible to recover?

Thanks,
Ricardo

**Comment by catgirlcrypto** [comment]
Hello, please can anyone help me? I have KSM stuck as it was sent to a polkadot address (via Ledger X). I have tried following these instructions for KSM recovery and it's not working. 

I have updated the Kusama app on my ledger device. The problem is that I cannot get the app to stay open so that I can click 10 times on Developed by: Zondax.ch as it keeps booting  me out of the Kusama app.

Can anyone advise what to do here? I really need to access me KSM urgently.

Thank you.
Margie

**Comment by osorio200** [comment]
Hi! Make sure the Ledger Live application is closed when you do the ten clicks. I had that problem too. You have to quit Ledger Live once you made sure it is updated, you won’t need to do anything with it anymore.

good luck!

**Comment by catgirlcrypto** [comment]
Thanks @osorio200 - I managed to get it into a recovered account. Now I'm having an issue transferring it back into a KSM account. 

I am getting this error message:

1010: Invalid Transaction: Transaction has a bad signature

Anyone know how to resolve this?
